### PR TITLE
[FEATURE] 184129153 Assign Leave application to leave approver

### PR DESCRIPTION
**Pull request by Talleyrand333**
## Is this a Feature, Chore or Bug?
- [*] Feature
- [] Chore
- [] Bug

## Clearly and concisely describe the feature, chore or bug.
Automatically assign a Todo to the leave approver

## Analysis and design (optional)
Analyse and attach the design documentation

## Solution description
Describe your code changes in detail for reviewers.
I added a hook that creates the assignment after the leave application is created

## Is there a business logic within a doctype?
    - [] Yes
    - [*] No

## Output screenshots (optional)
Post the output screenshots, if a UI is affected or added due to this feature.

## Areas affected and ensured
List out the areas affected by your code changes.
Leave Application form

## Is there any existing behavior change of other features due to this code change?
Mention Yes or No. If Yes, provide the appropriate explanation.
Auto assignment of leave application form

## Did you test with the following dataset?
- [] Existing Data
- [*] New Data

## Was child table created?
NO
    - [] is attachment required?
     NO
        did you test attachment
## Did you delete custom field?
    - [] Yes
    - [*] No
        If yes, did you write a delete patch?

## Is patch required?
- [] Yes
- [*] No
    ## Was the patch test?

## Which browser(s) did you use for testing?
  - [*] Chrome
  - [] Safari
  - [] Firefox

### CODE REVIEW

File: one_fm/overrides/leave_application.py [hunk 1]
```diff
@@ -6,11 +6,14 @@
...
 class LeaveApplicationOverride(LeaveApplication):
-
+    def after_insert(self):
+        self.assign_to_leave_approver()
+        
     def validate(self):
         validate_active_employee(self.employee)
         set_employee_name(self)
         self.validate_dates()
+        # self.assign_to_leave_approver()
```
Comment (pjamsheer) [inline one_fm/overrides/leave_application.py]: you can remove this commented line

File: one_fm/overrides/leave_application.py [hunk 2]
```diff
@@ -22,7 +25,26 @@ def validate(self):
...
+                        'allocated_to':self.leave_approver,
+                        'reference_type':'Leave Application',
+                        'description':f'Please note that leave application {self.name} is awaiting for your approval',
+                        'reference_name':self.name
+                    })
+                    todo_doc.insert()
+                    todo_doc.save()
+                except:
+                    
+                    frappe.log_error(frappe.get_traceback(),"Error assigning to User")
```
Comment (pjamsheer) [inline one_fm/overrides/leave_application.py]: `from frappe.desk.form.assign_to import add` We can make use of this implementation for assignment

Comment (Talleyrand333) [reply one_fm/overrides/leave_application.py]: @pjamsheer Thanks alot. I will quickly make the changes

**Review by mymi14s — APPROVED** [review]

**Comment by mymi14s** [comment]
@Talleyrand333 redolve conflict

**Pull request merged**